codegen/src/generator/mod.rs

```diff
@@ -228,26 +227,24 @@ fn generate_struct_fields<P>(service: &Service, shape: &Shape, shape_name: &str,
 
         let type_name = capitalize_first(member.shape.to_string());
 
-        lines.push("#[allow(unused_attributes)]".to_owned());
-        lines.push(format!("#[serde(rename=\"{}\")]", member_name));
-
-        lines.append(&mut protocol_generator.generate_additional_annotations(service, shape_name, &type_name));
-
-        if let Some(shape_type) = service.shape_type_for_member(member) {
-            if shape_type == ShapeType::Blob {
-                lines.push(
-                    "#[serde(
-                        deserialize_with=\"::serialization::SerdeBlob::deserialize_blob\",
-                        serialize_with=\"::serialization::SerdeBlob::serialize_blob\",
-                        default,
-                    )]".to_owned()
-                );
-            } else if shape_type == ShapeType::Boolean && !shape.required(member_name) {
-                lines.push("#[serde(skip_serializing_if=\"::std::option::Option::is_none\")]".to_owned());
+        if serde_attrs {
```

Comment:
Sorry for the confusing diff. The changes are:

- Indent the block inside `if serde_attrs { ... }`
- Remove the line that adds `"#[allow(unused_attributes)]"`